Generator/Sources/Generator/Template.swift

```diff
@@ -31,14 +31,9 @@ enum Template {
             return """
             // MARK: - Factory
             
-            struct Factory<T> {
+            struct Factory<T: Providable> {
```

Comment:
Compete with https://github.com/Nonchalant/FactoryProvider/pull/1
So, I will not support `Protocol` 🚀